test/infinity/Bridge/CMARK.RetireCMARK.t.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.8.16;
+
+import "../HelperContract.sol";
+import {RetireCMARKFacet} from "../../../src/infinity/facets/Bridges/CMARK/RetireCMARKFacet.sol";
+import {RetirementQuoter} from "../../../src/infinity/facets/RetirementQuoter.sol";
+import {LibRetire} from "../../../src/infinity/libraries/LibRetire.sol";
+import {LibTransfer} from "../../../src/infinity/libraries/Token/LibTransfer.sol";
+import {IERC20} from "oz/token/ERC20/IERC20.sol";
+
+import {console2} from "../../../lib/forge-std/src/console2.sol";
+
+import "../TestHelper.sol";
+import "../../helpers/AssertionHelper.sol";
+
+contract RetireCMARKFacetTest is TestHelper, AssertionHelper {
+    event CarbonRetired(
+        LibRetire.CarbonBridge carbonBridge,
+        address indexed retiringAddress,
+        string retiringEntityString,
+        address indexed beneficiaryAddress,
+        string beneficiaryString,
+        string retirementMessage,
+        address indexed carbonPool,
+        address carbonToken,
+        uint256 tokenId,
+        uint256 retiredAmount
+    );
+
+    RetireCMARKFacet retireCMARKFacet;
+    RetirementQuoter quoterFacet;
+    ConstantsGetter constantsFacet;
+
+    uint256 defaultCarbonRetireAmount = 5 * 1e18;
+    string beneficiary = "Test Beneficiary";
+    string message = "Test Message";
+    string entity = "Test Entity";
+    string consumptionCountryCode = "US";
+
+    address diamond = vm.envAddress("INFINITY_ADDRESS");
+    address beneficiaryAddress = vm.envAddress("BENEFICIARY_ADDRESS");
+
+    address CMARK = 0x103A806A88199ED11f16f1ECc7aC40fEFE12feb5;
+
+    function setUp() public {
+        addConstantsGetter(diamond);
+        constantsFacet = ConstantsGetter(diamond);
+        retireCMARKFacet = RetireCMARKFacet(diamond);
+        quoterFacet = RetirementQuoter(diamond);
+
+        upgradeCurrentDiamond(diamond);
+    }
+
+    function test_infinity_cmarkRetireExactCarbon() public {
+        uint256 tokenId = 1; // TODO: decide how to set this
+        uint256 retireAmount = 100e18;

```

Comment:
could fuzz here instead. I don't think it's essential in this case but it's a good sanity check.

Reply:
do we have any examples elsewhere in the repo of doing fuzzing?

Reply:
hm, guess not. i guess we are doing that routinely only in the marketplace repo.